rails_exercises/todo_app/config/routes.rb

```diff
@@ -0,0 +1,68 @@
+TodoApp::Application.routes.draw do
+  root to: 'task#list'
+
+  get 'task/delete_confirmation'
+  post 'task/complete'
+
+  # specifying delete and update as exceptions, since for it form submit is used
+  resources :task, except: [:update, :delete, :show]
+  post 'task/update'
+  post 'task/delete'
```

Comment:
The routing setup is not very conventional.
A more common way I think would be:
```ruby
resources :tasks do
  # This assumes
  # tasks#index - GET /tasks
  # tasks#show - GET /tasks/:id
  # tasks#new - GET /tasks/new
  # tasks#create - POST /tasks
  # tasks#update - PUT /tasks/:id
  # tasks#destroy - DELETE /tasks/:id
  member do
    get :delete_confirmation # GET /tasks/:id/delete_confirmation
    post :complete # POST /tasks/:id/complete
  end
end
```
Any reasons for a more custom setup?

Reply:
*Update* and  *Delete* were changed so that I can do form submit without extra Js, as it was requested. With that being, said I've adjusted it to member syntax I was not aware of, thank you for the suggestion.

Reply:
As we discussed, Rails should be able to work around this issue with its hidden `_method` parameter.